spectral_cube/lower_dimensional_structures.py

```diff
@@ -144,6 +144,11 @@ def __array_finalize__(self, obj):
         self._fill_value = getattr(obj, '_fill_value', np.nan)
         self._wcs_tolerance = getattr(obj, '_wcs_tolerance', 0.0)
 
+        if isinstance(obj, VaryingResolutionOneDSpectrum):
+            self._beams = getattr(obj, '_beams', None)
+        else:
+            self._beam = getattr(obj, '_beam', None)
```

Comment:
it might be better to have `VaryingResolutionOneDSpectrum` have an `__array_finalize__` that deletes the `_beam` object and adds the `_beams` object?  Or, have each subclass implement their own `__array_finalize__` that adds the appropriate type (after using super() to call this `finalize`)